File: backend/app/schemas/contract.py
from pydantic import BaseModel, Field
from datetime import datetime, UTC


class ContractPayload(BaseModel):
    session_id: str
    final_price: float = Field(..., gt=0)
    final_delivery_days: int = Field(..., gt=0)
    final_sla_percent: float = Field(..., gt=0, le=100)
    buyer_justification: str = ""
    vendor_justification: str = ""
    total_rounds: int = Field(..., ge=1)
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
    pdf_file_path: str = ""

File: frontend/src/types/contract.ts
export interface ContractPayload {
  id: string;
  session_id: string;
  final_price: number;
  final_delivery_days: number;
  final_sla_percent: number;
  pdf_file_path: string;
  created_at: string;
}

export interface ContractDetails {
  id: string;
  session_id: string;
  final_price: number;
  final_delivery_days: number;
  final_sla_percent: number;
  pdf_file_path: string;
  created_at: string;
}
